File: Src/Icm.Core/Math/LimitedQueue.vb
Namespace Icm.MathTools

    Public Class LimitedQueue(Of T)

        Private store_ As Generic.LinkedList(Of T)

        Private limit_ As Integer

        Public Sub New(ByVal limit As Integer)
            store_ = New Generic.LinkedList(Of T)
            limit_ = limit
        End Sub

        Public Sub Enqueue(ByVal o As T)
            If store_.Count = limit_ Then
                Dequeue()
            End If
            store_.AddLast(o)
        End Sub

        Public Sub Dequeue()
            store_.RemoveFirst()
        End Sub

        Public Function Head() As T
            Return store_.First.Value
        End Function

        Public Function Tail() As T
            Return store_.Last.Value
        End Function
    End Class
End Namespace

File: Src/Icm.Core/Math/LimitedQueue.cs
namespace Icm.MathTools
{

	public class LimitedQueue<T>
	{


		private Generic.LinkedList<T> store_;

		private int limit_;
		public LimitedQueue(int limit)
		{
			store_ = new Generic.LinkedList<T>();
			limit_ = limit;
		}

		public void Enqueue(T o)
		{
			if (store_.Count == limit_) {
				Dequeue();
			}
			store_.AddLast(o);
		}

		public void Dequeue()
		{
			store_.RemoveFirst();
		}

		public T Head()
		{
			return store_.First.Value;
		}

		public T Tail()
		{
			return store_.Last.Value;
		}
	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
